rewrite-java/src/main/java/org/openrewrite/java/ReplaceConstantWithAnotherConstant.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.Cursor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeUtils;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class ReplaceConstantWithAnotherConstant extends Recipe {
+
+    @Option(displayName = "Fully qualified name of the constant to replace", example = "org.springframework.http.MediaType.APPLICATION_JSON_VALUE")
+    String existingFullyQualifiedConstantName;
+
+    @Option(displayName = "Fully qualified name of the constant to use in place of existing constant", example = "org.springframework.http.MediaType.APPLICATION_JSON_VALUE")
+    String fullyQualifiedConstantName;
+
+    @Override
+    public String getDisplayName() {
+        return "Replace constant with another constant";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace constant with another constant, adding/removing import on class if needed.";
+    }
+
```

Comment:
Maybe it's good to add a applicability test that checks for the use of the owning type, such that the visitor isn't called for every field access and identifier.